### ISSUE TITLE
about training

### ISSUE THREAD

**Issue by RincolF** [slei109/PATNet#46]
Thanks for your work!
I'm getting the following reported error while training:

Traceback (most recent call last):
  File "..\PATNet-master\train.py", line 119, in <module>
    val_loss, val_miou, val_fb_iou = train(epoch, model, dataloader_val, optimizer, training=False)
  File "..\PATNet-master\train.py", line 55, in train
    average_meter.write_result('Training' if training else 'Validation', epoch)
  File "..\PATNet-master\common\logger.py", line 77, in write_result
    loss_buf = torch.stack(self.loss_buf)
RuntimeError: stack expects a non-empty TensorList

I have no idea why it does this at the end of training a epoch, which should go on to the next epoch

**Comment by llopen** [comment]
Hello! I also encountered the same error. Have you solved it？

**Comment by RincolF** [comment]
> Hello! I also encountered the same error. Have you solved it？

You need to build and set the val path correctly in train.py:
```
    # Dataset initialization
    FSSDataset.initialize(img_size=400, datapath=args.datapath)
    dataloader_trn = FSSDataset.build_dataloader(args.benchmark, args.bsz, args.nworker, args.fold, 'trn')
    FSSDataset.initialize(img_size=400, datapath='path_to_your_dataset')
    dataloader_val = FSSDataset.build_dataloader('fss', args.bsz, args.nworker, '0', 'val')
```
The default benchmark of dataloader_val is 'fss', which means using FSS-1000 dataset to validate your model while training. In this context, you should download fss-1000 and then configure train.py and maybe fss.py properly. In my case, I'm training my own dataset, so I split them into 'trn' and 'val', and the two folders have completely different classes.

**Comment by bowen099** [comment]
Thank you, I've solved the problem. Following the same approach as the author's experiment, I used Pascal for training and downloaded the FSS dataset as the validation set. After encountering the aforementioned error, I modified the path /home/xxxx/code/PATNet-master/fewshot_data/ab_wheel (where this part represents the specific category name) to /xxxx/code/PATNet-master/fewshot_data/FSS-1000/ab_wheel. In other words, I added a new folder named FSS-1000 under fewshot_data, and placed all category files within this folder. This adjustment successfully matched the code's reading method.
